src/main/java/seedu/multitasky/storage/XmlAdaptedEntry.java

```diff
@@ -43,35 +46,71 @@ public XmlAdaptedEntry() {
      * this will not affect the created XmlAdaptedEntry
      */
     public XmlAdaptedEntry(ReadOnlyEntry source) {
-        formatter.setLenient(false);
         name = source.getName().fullName;
-        startDateAndTime = formatter.format(source.getStartDateAndTime().getTime());
-        endDateAndTime = formatter.format(source.getEndDateAndTime().getTime());
+
+        if (source.getStartDateAndTime() != null) {
+            startDateAndTime = convertDateToString(source.getStartDateAndTime());
+        }
+        if (source.getEndDateAndTime() != null) {
+            endDateAndTime = convertDateToString(source.getEndDateAndTime());
+        }
         tagged = new ArrayList<>();
         for (Tag tag : source.getTags()) {
             tagged.add(new XmlAdaptedTag(tag));
         }
     }
 
+    public String convertDateToString(Calendar given) {
+        String dateToString = df.format(given.getTime());
+        return dateToString;
+    }
+
+    public Calendar convertStringToDate(String given) throws NullPointerException {
+        Calendar setDate = null;
+        Date toConvert = new Date();
+        try {
+            toConvert = df.parse(given);
+        } catch (ParseException e) {
+            System.out.println("Invalid date!");
```

Comment:
was talking about this part, should throw some sort of exception here instead of println